pjsip/src/pjsip/sip_transport.c

```diff
@@ -1079,8 +1087,21 @@ static void transport_idle_callback(pj_timer_heap_t *timer_heap,
      * race condition with pjsip_tpmgr_acquire_transport2().
      */
     pj_lock_acquire(tp->tpmgr->lock);
+
     if (pj_atomic_get(tp->ref_cnt) == 0) {
         tp->is_destroying = PJ_TRUE;
+
+        if (entry_id == INITIAL_IDLE_TIMER_ID) {
+            if (tp->last_recv_len > 0 && tp->tpmgr->tp_drop_data_cb) {
+                pjsip_tp_dropped_data dd;
+                pj_bzero(&dd, sizeof(dd));
+                dd.tp = tp;
+                dd.data = NULL;
+                dd.len = tp->last_recv_len;
+                dd.status = PJ_ESOCKETSTOP;
+                (*tp->tpmgr->tp_drop_data_cb)(&dd);
+            }
+        }
```

Comment:
add log that transport is about to be destroyed due to idle, for both idle timers: initial and consecutive?